diesel/src/pg/connection/result.rs

```diff
@@ -74,14 +70,8 @@ impl PgResult {
                         _ => DatabaseErrorKind::Unknown,
                     };
                 let error_information = Box::new(PgErrorInformation(internal_result));
-                // NOTE: Ideally, we'd remove this check, in favor of matching
-                // on an error code (instead of the more brittle description
-                // string).
-                //
-                // Unfortunately, this particular error is often propagated in
-                // cases - such as network disconnect - without any accompanying
-                // "SqlState", and therefore without an error code..
-                if error_information.message() == CLOSED_CONNECTION_MSG {
+                let pg_connection: *const PGconn = conn.into();
+                if unsafe { PQstatus(pg_connection) } == ConnStatusType::CONNECTION_BAD {
```

Comment:
We should likely add a method to `RawConnection` to wrap the call to `PQstatus` there, so something like `RawConnection::get_status() -> i32` or so. This has multiple advantages:
* We can only call `PQstatus` with a valid connection object
* We have most connection related unsafe code in one module.

Reply:
Thanks for the review. And this feedback makes a ton of sense. Wasn't quite sure if this next commit addresses things quite the way you envisioned but will push it up now anyway :)